packages/graph/hash_graph/lib/graph/src/store/postgres/mod.rs

```diff
@@ -345,17 +345,14 @@ where
                 .change_context(UpdateError));
         }
 
-        let base_uri = uri.base_uri();
-
         // TODO - address potential race condition
         //  https://app.asana.com/0/1202805690238892/1203201674100967/f
+        let previous_ontology_type =
+            <Self as Read<T::WithMetadata>>::read_one(self, &Filter::for_base_uri(uri.base_uri()))
```

Comment:
Somehow, Rust fails to detect, that we don't want to read an `Entity` but `T::WithMetadata`, so this specifies the trait explicitly